x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/add_entry_button.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiIcon,
+  EuiPopover,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+} from '@elastic/eui';
+import React, { useCallback, useState } from 'react';
+import * as i18n from './translations';
+
+interface Props {
+  isDocumentAvailable?: boolean;
+  isIndexAvailable?: boolean;
+  onDocumentClicked?: () => void;
+  onIndexClicked?: () => void;
+}
+
+export const AddEntryButton: React.FC<Props> = React.memo(
+  ({
+    isDocumentAvailable = true,
+    isIndexAvailable = true,
+    onDocumentClicked,
+    onIndexClicked,
+  }: Props) => {
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+    const closePopover = useCallback(() => setIsPopoverOpen(false), []);
+    const handleIndexClicked = useCallback(() => {
+      closePopover();
+      onIndexClicked?.();
+    }, [closePopover, onIndexClicked]);
+
+    const handleDocumentClicked = useCallback(() => {
+      closePopover();
+      onDocumentClicked?.();
+    }, [closePopover, onDocumentClicked]);
+
+    const onButtonClick = useCallback(() => setIsPopoverOpen((prevState) => !prevState), []);
+
+    const items = [
+      <EuiContextMenuItem
+        aria-label={i18n.INDEX}
+        key={i18n.INDEX}
+        icon="index"
+        onClick={handleIndexClicked}
+        disabled={!isIndexAvailable}
+      >
+        {i18n.INDEX}
+      </EuiContextMenuItem>,
+      <EuiContextMenuItem
+        aria-label={i18n.DOCUMENT}
+        key={i18n.DOCUMENT}
+        icon="document"
+        onClick={handleDocumentClicked}
+        disabled={!isDocumentAvailable}
+      >
+        {i18n.DOCUMENT}
+      </EuiContextMenuItem>,
+    ];
+    return onIndexClicked || onDocumentClicked ? (
+      <EuiPopover
+        button={
+          <EuiButton iconType="arrowDown" iconSide="right" onClick={onButtonClick}>
+            <EuiIcon type="plusInCircle" />
+            {i18n.NEW}
```

Comment:
@bojanasan, I updated this to say `New` to match the latest mocks (which match the unification mocks as well), but I noticed the other add buttons on the `Prompts` settings say the asset name instead of `New`? 

Should I update this to say `+ Entry`, or update the others to say `+ New`?

<img width="1042" alt="image" src="https://github.com/user-attachments/assets/6c0b29b7-8098-4ff1-9dbf-af2950fc3b8c">
<img width="1045" alt="image" src="https://github.com/user-attachments/assets/e44bea37-e1cc-4528-9d4f-b4f5a116a34a">
<img width="1039" alt="image" src="https://github.com/user-attachments/assets/0845719e-09e3-47ce-8261-0ea15625d952">